phoenix-core-client/src/main/java/org/apache/phoenix/util/TupleUtil.java

```diff
@@ -210,4 +219,35 @@ public static int write(Tuple result, DataOutput out) throws IOException {
           }
         return size;
     }
+
+    /**
+     * Convert the given Tuple containing list of Cells to ResultSet with similar effect as if
+     * SELECT * FROM <table-name> is queried.
+     *
+     * @param toProject Tuple to be projected.
+     * @param tableName Table name.
+     * @param conn Phoenix Connection object.
+     * @return ResultSet for the give single row.
+     * @throws SQLException If any SQL operation fails.
+     */
+    public static ResultSet getResultSet(Tuple toProject, String tableName, Connection conn)
+        throws SQLException {
+        try (PhoenixResultSet resultSet = (PhoenixResultSet) conn.createStatement()
+            .executeQuery("SELECT * FROM " + tableName)) {
```

Comment:
For my understanding, will this query actually execute? If yes, should we add a `LIMIT 1`?

Reply:
Got it, since the real execution of it happens during a `next` call?

Reply:
Yes and also we are using Prefetched ResultSet so we are providing Tuple already. The new ResultSet does not even have ResultIterator so no real scan will happen.